pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -56,7 +58,7 @@
 /**
  * This class implements the Context interface exposed to the user.
  */
-class ContextImpl implements Context {
+class ContextImpl implements Context, SinkContext, SourceContext {
```

Comment:
does it make sense to have a base interface as `Context` and other contexts can implements it
such as: `SinkContext`, `SourceContext`, `FunctionContext`?

Reply:
well Context is the "function" context and we can't really change that now our we will break backwards compatibility. 

The APIs SinkContext, SourceContext, and Context is and will be different.  The three won't have the all the same APIs but currently we are only exposing one method in SinkContext and SourceContext thus we can just use ContextImpl but in the future that might change